src/coreclr/vm/reflectioninvocation.cpp

```diff
@@ -617,7 +617,7 @@ extern "C" void QCALLTYPE RuntimeMethodHandle_InvokeMethod(
         // We have a special case for Strings...The object is returned...
         if (fCtorOfVariableSizedObject) {
             PVOID pReturnValue = &callDescrData.returnValue;
-            gc.retVal = *(OBJECTREF *)pReturnValue;
+            gc.retVal = ObjectToOBJECTREF(*(Object**)pReturnValue);
```

Comment:
Apply recommended fix from 
https://github.com/dotnet/runtime/blob/215a79a3f37a3afec0ac2c418b6fbeacb26e1585/src/coreclr/vm/object.cpp#L1135-L1136

Reply:
Should we search the codebase for these pattern? `*(OBJECTREF*)` and `*(OBJECTREF *)`.

I see candidates in `comtoclrcall.cpp` in `InvokeStub()` and `CLRToCOMEventCallWorker()`. Something suspect in `EEDbgInterfaceImpl::GetObjectFromHandle()`. `CopyOBJECTREFToStack()` in `callsiteinspect.cpp` has two. In `interpexec.cpp`, there is something to consider under `INTOP_LDELEM_REF`.